### PR TITLE
Add argument to directly control the size of the initial design

### PR DESCRIPTION
**Pull request by mfeurer**
This might be advantageous in cases where one does meta-learning and wants to ensure that exactly 2 or 3 configurations are used to start.

### CODE REVIEW

File: smac/initial_design/initial_design.py [hunk 1]
```diff
@@ -82,6 +83,8 @@ def __init__(self,
             specify if target algorithm has to be run once on the first configuration before the intensify call
         fill_random_configs: bool
             fill budget with random configurations if initial incumbent sampling returns only 1 configuration
+        init_budget : int, optional
+            Maximal initial budget (alternative to ``n_configs_x_params``)
```
Comment (KEggensperger) [inline smac/initial_design/initial_design.py]: It is unclear from this documentation what happens if both or None are given

Comment (KEggensperger) [reply smac/initial_design/initial_design.py]: It is also unclear what happens if, e.g. n_configs_x_params is higher/lower than the initial budget

File: test/test_initial_design/test_initial_design.py [hunk 2]
```diff
@@ -22,8 +22,12 @@ def setUp(self):
...
             name="x1", lower=1, upper=10, default_value=1)
         )
-        self.scenario = Scenario({'cs': self.cs, 'run_obj': 'quality',
-                                  'output_dir': ''})
+        self.scenario = Scenario({
+            'cs': self.cs,
+            'run_obj': 'quality',
+            'output_dir': '',
+            'ta_run_limit': 100,
+        })
```
Comment (KEggensperger) [inline test/test_initial_design/test_initial_design.py]: Does this change anything?

Comment (mfeurer) [reply test/test_initial_design/test_initial_design.py]: No, it is only necessary to test that the exception is triggered if we request an initial design larger than the number of target runs.

File: smac/initial_design/initial_design.py (hunk 1 above)
Comment (mfeurer) [reply smac/initial_design/initial_design.py]: I just extended the docstrings to clarify the behavior.

**Pull request merged**